### DISCUSSION TITLE
Links & line-wrap

### DISCUSSION THREAD

**Discussion by MollyMaclachlan** [TomSchimansky/CustomTkinter#223] [Q&A]
There a couple of things I'd like to be able to do that would make my life a lot easier:

- First is to create hyperlinks within labels. I tried to do this using a method recommended by several people at [this StackOverflow question](https://stackoverflow.com/questions/11949391/how-do-i-use-tkinter-to-create-line-wrapped-text-that-fills-the-width-of-the-win), but clicking on the "link" did nothing. I imagine this is probably because they're working with basic Tkinter labels whereas I'm working with CustomTkinter ones. If anyone knows if there's a way to do this, it'd be really nice.
- Second is line-wrap within labels. I have some pretty long labels on my programm and it's annoying to have to manually insert the `\n`s so that all the text is visible. It'd be a real time saver if there were a way to automatically wrap the lines, but I haven't been able to find one.

Thanks in advance for whomever reads/answers this!

**Comment by felipetesc** [comment]
Hi. To create hiperlinks I think the best is to use buttons for now. After trying several things I came to that ( it's ugly though - but I hope it can help to solve the issue later on)

By the way, the problem about the code below is based on the fact that:
- we need to copy font attributes from the button and pass them to text_label inside the methods focus_in_callback and focus_out_callback
- we are accessing text_label which should not happen directly

```python
import tkinter
import customtkinter
from customtkinter import ThemeManager
DARK_MODE = "dark"
customtkinter.set_appearance_mode(DARK_MODE)
customtkinter.set_default_color_theme("blue")


def focus_in_callback(event):
    print("in")
    event.widget.config(fg='Blue')
    event.widget.config(font=('Roboto',12,'underline'))


def focus_out_callback(event):
    print("out")
    event.widget.config(fg='Black')
    event.widget.config(font=('Roboto',12))


class App(customtkinter.CTk):

    def __init__(self):
        super().__init__()
        self.title("")
        self.geometry("800x600")

        lb_with_link = customtkinter.CTkButton(self, fg_color=None, hover_color=None, text="Hello", cursor="hand2")

        lb_with_link.text_label.bind("<Enter>", focus_in_callback)
        lb_with_link.text_label.bind("<Leave>", focus_out_callback)
        lb_with_link.pack()

a = App()
a.mainloop()

```
Suggestion: to change CTkButton, or to create CTkLink class, or to extend CTkButton

If CTkButton is changed (ads complexity to CTkButton and perhaps it is not as fast as it can be):
CTkButton could have a boolean field like  'isLink', or a str field like 'html',  or something like this. 
If it has a boolean field is_link:
- the methods on_enter and on_leave could be changed to add the logic to change the font appereance and allow the text to be underlined
- cursor can have a default like hand2 if is_link is true
- inside init the bg color should have a default like None, so it inherits the parent bg if is_link is true

If a class like CTkLink is created (less complexity and perhaps the render is a faster- **I belive this is the best idea**):
- the class constructor does'nt have text attribute anymore but html
- it can have url_hover_bg attribute
- perhaps background color can be ommited

If CTkButton is extended (less complexity than enhaencing CTkButton and perhaps this implementation provides the slower drawing rendering mechanism):
-overide all needed methods

**Reply by Yug-Desai** [reply]
Hey I am quite new to python. I have created a simple login form however I would like to use a label at the bottom of the GUI (which has the text "Register and create an account.") 
I want this label to link to a new GUI (top level) but I don't know how. Is there a way I can use a command to give the label so that when i click it, it opens a new GUI.
I know how to do it with a Button but its different I see, with a label, is it possible to do so?
